src/dags/overlastgebieden.py

```diff
@@ -64,66 +56,164 @@ def checker(records, pass_value):
+            task_id=f"download_{file}",
+            # Default swift = Various Small Datasets objectstore
+            swift_conn_id="SWIFT_DEFAULT",
+            container="overlastgebieden",
+            object_id=f"{file}",
+            output_path=f"{tmp_dir}/{file}",
         )
+        for file in files_to_download
+    ]
 
-    extract_shp = BashOperator(
-        task_id="extract_shp",
-        bash_command=f"ogr2ogr -f 'PGDump' -t_srs EPSG:28992 -skipfailures -nln {dag_id}_new "
-        f"{tmp_dir}/{dag_id}.sql {tmp_dir}/OOV_gebieden_totaal.shp",
-    )
+    # 4. Dummy operator acts as an interface between parallel tasks to another parallel tasks with different number of lanes
+    #  (without this intermediar, Airflow will give an error)
+    Interface = DummyOperator(task_id="interface")
+
+    # 5. Create SQL
+    SHP_to_SQL = [
+        BashOperator(
+            task_id=f"create_SQL_{key}",
+            bash_command=f"ogr2ogr -f 'PGDump' "
+            f"-s_srs EPSG:28992 -t_srs EPSG:28992 "
+            f"-nln {dag_id}_{key}_new "
+            f"{tmp_dir}/{dag_id}_{key}_new.sql {tmp_dir}/OOV_gebieden_totaal.shp "
+            f"-lco GEOMETRY_NAME=geometry "
+            f"-nlt PROMOTE_TO_MULTI "
+            f"-lco precision=NO "
+            f"-lco FID=id "
+            f"-sql \"SELECT * FROM OOV_gebieden_totaal WHERE 1=1 AND TYPE = '{code}'\" ",
+        )
+        for key, code in tables_to_create.items()
+    ]
 
-    convert_shp = BashOperator(
-        task_id="convert_shp",
-        bash_command=f"iconv -f iso-8859-1 -t utf-8  {tmp_dir}/{dag_id}.sql > "
-        f"{tmp_dir}/{dag_id}.utf8.sql",
-    )
+    # 6. Create TABLE
+    create_tables = [
+        BashOperator(
+            task_id=f"create_table_{key}",
+            bash_command=f"psql {pg_params()} < {tmp_dir}/{dag_id}_{key}_new.sql",
+        )
+        for key in tables_to_create.keys()
+    ]
 
-    create_tables = PostgresFilesOperator(
-        task_id="create_tables", sql_files=[f"{tmp_dir}/{dag_id}.utf8.sql"],
+    # 7. Rename COLUMNS based on Provenance
+    provenance_translation = ProvenanceRenameOperator(
+        task_id="rename_columns",
+        dataset_name=f"{dag_id}",
+        prefix_table_name=f"{dag_id}_",
+        postfix_table_name="_new",
+        rename_indexes=False,
+        pg_schema="public",
     )
 
-    process_table = PostgresOperator(task_id="process_table", sql=PROCESS_TABLE,)
+    # 8. Remove invalid geometry records
+    # the source has some invalid records where the geometry is not present (NULL)
+    # or the geometry in itself is not valid
+    # the removal of these records (less then 5) prevents errorness behaviour
+    # to do: inform the source maintainer
+    remove_null_geometry_records = [
+        PostgresOperator(
+            task_id=f"remove_null_geom_records_{key}",
+            sql=[
+                f"DELETE FROM {dag_id}_{key}_new WHERE 1=1 AND (geometry IS NULL OR ST_IsValid(geometry) is false); COMMIT;",
```

Comment:
Er is in Postgresql zo'n zelf-reparatie mogelijkheid op invalide gemetrieën, heb je die er nog geprobeerd erop los te laten?

Reply:
Ah! Dat werkt goed, ST_MakeValid. Dat wist ik niet. Super ik voeg deze toe.